### ISSUE TITLE
[TF2.4rc2] linalg.svd missing kernel for DT_HALF

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/tensorflow#45136]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab (Ubuntu)
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.4.0rc2
- Python version: 3.6 (probably all)

**Describe the current behavior**
tf.linalg.svd fails on tf.half dtypes in TF2.4
https://colab.research.google.com/drive/1Y3FRV5qmQnnGUVHBX2wsFa0hC2PVDUON?usp=sharing
```
NotFoundError: Could not find device for node: {{node Svd}} = Svd[T=DT_HALF, compute_uv=true, full_matrices=false]
All kernels registered for op Svd:
  device='GPU'; T in [DT_DOUBLE]
  device='GPU'; T in [DT_FLOAT]
  device='CPU'; T in [DT_FLOAT]
  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_COMPLEX64]
  device='CPU'; T in [DT_COMPLEX128]
 [Op:Svd]
```

**Describe the expected behavior**
tf.linalg.svd works on tf.half dtypes in TF2.3
https://colab.research.google.com/drive/1Gf_C6SAQgUHzEQfkZPXD8PITL8lW17-I?usp=sharing

**Comment by amahendrakar** [comment]
Was able to reproduce the issue. Code runs without any issues on [TF v2.3](https://colab.research.google.com/gist/amahendrakar/f88cb994811e3f3b1c0cbedf90831e84/45136-2-3.ipynb). 

However, on running with [TF v2.4.0rc2](https://colab.research.google.com/gist/amahendrakar/db9d1b97d70ec1a2b51d1055f02bb739/45136-2-4.ipynb) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/35974fde861cb25d739bc80c91e03911/45136-2-5.ipynb#scrollTo=3PSlWW6GcMFu) it throws an error stating `NotFoundError: Could not find device for node`. Please find the attached gist. Thanks!

**Comment by WindQAQ** [comment]
@seanpmorgan We can either fix it by decorating SVD with `tf.function(experimental_compile=True)` in addons or specialize float16 kernel for SVD. It makes no sense to me when ops allow float16, but kernel does not specialize it. Therefore, I prefer option 2, but I'm afraid that it cannot meet 2.4.0 stable version release unless some owners can cherrypickup it.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/linalg/svd_op_gpu.cu.cc#L431-L436
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/linalg_ops.cc#L548

**TL;DR**

In TF2.4, XLA_{CPU,GPU} are disabled by default. See the following lines or verify with `tf.config.list_logical_devices`.

https://github.com/tensorflow/tensorflow/blob/r2.3/tensorflow/compiler/jit/flags.cc#L162
https://github.com/tensorflow/tensorflow/blob/r2.4/tensorflow/compiler/jit/flags.cc#L162
https://github.com/tensorflow/tensorflow/commit/68016e2697820763096254bfb139478f40c79442

Because soft device placement is enabled by default and float16 SVD kernel is registered for XLA, SVD thus can work in TF2.3 without float16 registeration on CPU/GPU.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/xla_op_registry.h#L48-L49
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/kernels/xla_svd_op.cc#L96-L97

**Comment by rmothukuru** [comment]
@WindQAQ,
Do you mean to say the behavior is as expected? Can we close this issue? Thanks!

**Comment by sushreebarsa** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
